packages/xstate-immer/src/index.ts

```diff
@@ -16,37 +16,51 @@ export interface ImmerAssignAction<TContext, TEvent extends EventObject>
   assignment: ImmerAssigner<TContext, TEvent>;
 }
 
-export function assign<TContext, TEvent extends EventObject = EventObject>(
+function immerAssign<TContext, TEvent extends EventObject = EventObject>(
   assignment: ImmerAssigner<TContext, TEvent>
 ): AssignAction<TContext, TEvent> {
-  // @ts-ignore (possibly infinite TS bug)
   return xstateAssign((context, event) => {
     return produce(context, (draft) => void assignment(draft, event));
   });
 }
 
-interface PatchEventObject extends EventObject {
-  patches: ReturnType<typeof produceWithPatches>;
+export { immerAssign as assign };
+
+export interface ImmerUpdateEventObject extends EventObject {
+  input: any;
 }
 
-export function patchEvent<TContext>(
-  type: string,
-  context: TContext,
-  recipe: (draftContext: TContext) => void
-): PatchEventObject {
-  return {
-    type,
-    patches: produceWithPatches(context, recipe)
-  };
+export interface ImmerUpdateEvent<TType extends string, TInput = any> {
```

Comment:
Is this used anywhere? I dont see much difference between this and the one above it. 

Also - should we default second type param to any? It makes it less type-safe by default

Reply:
Yes, check the tests. It's meant to make defining these events simpler:

```ts
type SomeEvent = ImmerUpdateEvent<"UPDATE_NAME", string>;

// rather than...

type SomeEvent = { type: "UPDATE_NAME"; input: string }

// which is more mistake-prone
```

Reply:
Ok, I see - it seems that the other one can just be removed. Take a look here: https://github.com/davidkpiano/xstate/pull/1130/files